File: 05-may/26-count-the-number-of-special-characters-i/solution.java
class Solution {
    public int numberOfSpecialChars(String word) {
        boolean[] lowercase = new boolean[26];
        boolean[] uppercase = new boolean[26];

        for (int i = 0; i < word.length(); i++) {
            char ch = word.charAt(i);
            if (ch >= 'a' && ch <= 'z') {
                lowercase[ch - 'a'] = true;
            } else {
                uppercase[ch - 'A'] = true;
            }
        }

        int answer = 0;
        for (int i = 0; i < 26; i++) {
            if (lowercase[i] && uppercase[i]) {
                answer++;
            }
        }

        return answer;
    }
}

File: 05-may/26-count-the-number-of-special-characters-i/solution.js
/**
 * @param {string} word
 * @return {number}
 */
var numberOfSpecialChars = function(word) {
    const lowercase = new Set();
    const uppercase = new Set();

    for (const ch of word) {
        if (ch >= 'a' && ch <= 'z') {
            lowercase.add(ch);
        } else {
            uppercase.add(ch.toLowerCase());
        }
    }

    let answer = 0;
    for (const ch of lowercase) {
        if (uppercase.has(ch)) {
            answer++;
        }
    }

    return answer;
};
